internal/scraper/scraper_test.go

```diff
@@ -0,0 +1,457 @@
+				},
+			},
+		},
+
+		"http": {
+			prober: prober.ProbeHTTP,
+			setup: func(*testing.T) (string, func()) {
+				httpSrv := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+					w.WriteHeader(http.StatusOK)
+				}))
+				// httpSrv.EnableHTTP2 = true
+				httpSrv.Start()
+				return httpSrv.URL, httpSrv.Close
+			},
+			config: bbeconfig.Module{
+				Prober:  "http",
+				Timeout: 2 * time.Second,
+				HTTP: bbeconfig.HTTPProbe{
+					IPProtocol:         "ip4",
+					IPProtocolFallback: false,
+					HTTPClientConfig:   promconfig.HTTPClientConfig{},
+				},
+			},
+		},
+
+		"http_ssl": {
+			prober: prober.ProbeHTTP,
+			setup: func(*testing.T) (string, func()) {
+				httpSrv := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+					w.WriteHeader(http.StatusOK)
+				}))
+				// httpSrv.EnableHTTP2 = true
+				httpSrv.StartTLS()
+				return httpSrv.URL, httpSrv.Close
+			},
+			config: bbeconfig.Module{
+				Prober:  "http",
+				Timeout: 2 * time.Second,
+				HTTP: bbeconfig.HTTPProbe{
+					IPProtocol:         "ip4",
+					IPProtocolFallback: false,
+					HTTPClientConfig: promconfig.HTTPClientConfig{
+						TLSConfig: promconfig.TLSConfig{
+							InsecureSkipVerify: true,
+						},
+					},
+				},
+			},
+		},
+
+		"dns": {
+			prober: prober.ProbeDNS,
+			setup: func(*testing.T) (string, func()) {
+				dnsSrv, dnsAddr := startDNSServer(":0", "udp6", recursiveDNSHandler)
+				go func() {
+					err := dnsSrv.ActivateAndServe()
+					if err != nil {
+						// this should never happen, but log it if it does
+						t.Log(err)
+					}
+				}()
+				return dnsAddr.String(), func() {
+					err := dnsSrv.Shutdown()
+					if err != nil {
+						// this should never happen, but log it if it does
+						t.Log(err)
+					}
+				}
+			},
+			config: bbeconfig.Module{
+				Prober:  "dns",
+				Timeout: 2 * time.Second,
+				DNS: bbeconfig.DNSProbe{
+					IPProtocol:         "ip4",
+					IPProtocolFallback: true,
+					QueryName:          "example.com",
+					TransportProtocol:  "udp",
+				},
+			},
+		},
+
+		"tcp": {
+			prober: prober.ProbeTCP,
+			setup: func(t *testing.T) (string, func()) {
+				ln, err := net.Listen("tcp4", ":0")
+				if err != nil {
+					t.Fatalf("Error listening on socket: %s", err)
+				}
+
+				ch := make(chan (struct{}))
+
+				go func() {
+					conn, err := ln.Accept()
+					if err != nil {
+						panic(fmt.Sprintf("Error accepting on socket: %s", err))
+					}
+					defer func() {
+						conn.Close()
+						close(ch)
+					}()
+
+					buf := make([]byte, 0, 1024)
+					_, _ = conn.Read(buf[:])
+				}()
+
+				return ln.Addr().String(), func() {
+					<-ch
+					ln.Close()
+				}
+			},
+			config: bbeconfig.Module{
+				Prober:  "tcp",
+				Timeout: 2 * time.Second,
+				TCP: bbeconfig.TCPProbe{
+					IPProtocol:         "ip4",
+					IPProtocolFallback: true,
+				},
+			},
+		},
+
+		"tcp_ssl": {
+			prober: prober.ProbeTCP,
```

Comment:
nit: let's move this as well, and maybe we can reuse the tcp setup method

Reply:
done